msg-socket/src/sub/stats.rs

```diff
@@ -84,3 +84,42 @@ impl SessionStats {
         self.latency.load(Ordering::Relaxed)
     }
 }
+
+#[derive(Debug, Default)]
+pub struct SocketWideStats {
```

Comment:
Agreed with having socket-agnostic stats for shared metrics, but I would like to see it implemented differently, something like the following:

```rs
struct SocketStats<S> {
   // ... socket-wide stats, and
   substats: S,
}
```

Where `S` = a stats struct specific to the socket implementation, for example `SubStats` or `PubStats`.

Reply:
Hi, @mempirate yeah a unified structure makes sense. Just want to clarify though: 
 Are you thinking of a struct like maybe `SocketStats<SpecificStats>` (eg. `SocketStats`<SubStats>) where specific `SpecificStats` (like `SubStats`) hold the implementation specific details like the session map and the new counters like `dropped_messages_total` for Sub? And for this specific PR should I only refactor the `SubSocket` stats into this new pattern, leaving the refactor of Pub/Rep/Req stats for future work ? I'm happy to do a big refactor if thats preferred!

Reply:
Yeah that's right! And would be great if you'd want to do this across the different socket types so we don't get any inconsistencies